sympy/matrices/matrices.py

```diff
@@ -1333,11 +1333,23 @@ def jordan_form(self, calc_transform=True, **kwargs):
         mat = self
         has_floats = any(v.has(Float) for v in self)
 
+        def dps(expr):
+            C = log(10)/log(2)
+            return max(1, int(round((int(expr) / C - 1).evalf())))
+
+        if has_floats:
+            max_prec = 0
+            for term in self._mat:
+                if isinstance(term, Float):
+                    max_prec = max(max_prec, term._prec)
```

Comment:
You can do `max_prec = max(term._prec for term in self._mat if isinstance(term, Float))`